BadIdeaFactory/statnews-buildfire
# Add initial version of RSS plugin

This initial version of the RSS plugin adds:

- A control settings area for app owners to set authenticated feed credentials
- Attempting to load and validate user data set by the SSO plugin
- Attempting to load and use authenticated feed credentials set by the control
- Negotiating which feed to use based on authentication status
- Fetching and rendering headlines appropriately

It is missing:

- The ability to read an article
- Styling/branding
- Live verification of the user information read from `localStorage` (i.e., ensuring the user is still an active user by checking the Auth0 API)
- Graceful handling of incorrect authenticated feed credentials
- Complete JSDocs of application JS

#### Steps to Test

**Prerequisites:** You'll need the username and password for the STAT Plus authenticated feed.

1. Pull the plugin and load it into a BuildFire SDK `plugins/` directory.
2. Start the SDK with `yarn start`
3. Navigate to the plugin tester (should be [this URL](http://localhost:3000/pluginTester/index.html#/plugin/rssPlugin))
4. **You should now get the unauthenticated "STAT News Headlines" in the widget.** ✋ Note: You'll see some individual entries prefixed "STAT Plus". These are unpaywalled STAT Plus articles inserted into the standard feed.

For the next steps (testing authenticated feed), you have to have the SSO Plugin (#5). Assuming you have that setup also, authenticate using that plugin and then come back to this plugin and continue:

1. Upon initial load, you should get a `console.error` about missing authenticated feed credentials (which it complains about since it recognizes you're an authenticated user), but the unauthenticated STAT News Headlines should still load as a fallback.
2. Enter valid authenticated feed credentials (from Prerequisites) in the left-hand ("Control Settings") section. Save and refresh.
3. **You should now get the authenticated "STAT Plus News Headlines" in the widget.**

#### Related Issues

Affects #4 (but does not yet resolve)
